### ISSUE TITLE
Issues with Demo Server

### ISSUE THREAD

**Issue by sirajc** [ng-bootstrap/core#278]
There is no guide available to setup the demo server. Found build:demo and demo-server tasks both of which are failing

* prism themes are not found by webpack
```
Module not found: Error: Cannot resolve module 'prismjs/themes/prism-okaidia.css' in D:\RnD\angular2\ng-bootstrap\core\demo\src\app
 @ ./demo/src/app/app.component.ts 15:0-43
```

* Console shows errors for angular and rxjs files from `node_modules`
* Console shows errors for typings files as well

* Typescript version is 1.6.2
* node_modules` are not excluded in tsconfig.json which slows the build, adding exclude makes it fast. This also slows down VSCODE

**Comment by sirajc** [comment]
@pkozlowski-opensource  Need to work on #264, but thought of completing setup before starting with discussions

PS: I can work on upgrading Typescript and other packages, update tsconfig as per guidelines, update gulpfile if required, but I am not much experienced in Webpack.

**Comment by icfantv** [comment]
@sirajc, the demo site IS working but is undergoing a _very large_ overhaul as we gear up for RC0.  as such this is not something we are going to spend time supporting.  I see you're on Windows and thus, are probably going to be pretty much on your own here as the rest of the team is on macs.

That said, make sure you blow away your `typings` and `node_modules` directories and do a fresh & clean install via `typings install` and `npm install` using the latest versions of typings and node.  You can run the demo server via `gulp demo-server` but again, I've got several PRs that completely overhaul this thing and we are in the middle of a team discussion about design.

You're also going to want the latest version of typescript.  Please look at the `package.json` file for the versions of things we are using.

**Comment by sirajc** [comment]
I had cleared both typings and node_modules and did fresh install.  
Do have a look at package.json again in master, it still refers to 1.6.2. `"typescript": "^1.6.2",`  you might want to upgrade this @icfantv 

I have waited for quite few months and can wait few more days untill things are settled out.  
I came here post discussion with @pkozlowski-opensource , who assured that thinge have changed and contributions are welcome

**Comment by sirajc** [comment]
Though issue is closed you might want to consider either add files or excludes config in tsconfig.json to improve build times and editor performance 
Something like this https://github.com/sirajc/angular2-bs4-navbar/blob/master/src/tsconfig.json

**Comment by icfantv** [comment]
Apologize about the TS version - I was under the impression that it had been upgraded.  My bad.  I'll submit a PR to update typescript and exclude `node_modules` but as I said, that you're on Windows is the most likely culprit as my build yields no errors and completes in about 8s including running the tests.  If you have access to a Mac, I would suggest using it, if nothing else, to see if this is indeed the case.  If not, you might try setting up a cygwin environment as that will, at the very least, fix your path issues.

I do not mean/want to deter you from submitting quality PRs, but I would ask that you look over the current state of the project before filing issues as had you done so, you would see the current WIP and remaining work needing to be done.  Also, the team is in a _very_ highly motivated state to release RC0 so we're not intending to wait around for things to just happen.

**Comment by sirajc** [comment]
Thanks @icfantv 
I use Git Bash which is basically like Cygwin. I raised this issue to know about the configuration that needs to be done since we didnt had docs
I tried and will try again to figure out and fix all the issues that crop up, will share the fixes here

One thing I can't get my head around is the prism theme, there is no prism dependency in package.json, we have prism loader but that is not used in webpack demo, so there is no way this dependency can be fulfilled. Just need help on this, rest all, I will figure out

**Comment by icfantv** [comment]
@sirajc, we committed a bunch of stuff today so please be sure to do a pull.

**Comment by sirajc** [comment]
+1 for typescript and exclusion of node_modules, All issues gone except prismjs one
Did `npm install prismjs` and voila demosite it up

@icfantv you might want to add a dependency in package.json for prismjs, `"prismjs": "^1.5.1",`
We have `import 'prismjs/themes/prism-okaidia.css';` in `app.component.ts` so obviously we need that dependency

**Comment by sirajc** [comment]
#290

**Comment by icfantv** [comment]
This is starting to feel like we're doing all sorts of unnecessary things just to accommodate building on Windows and so I'm going to defer to @foxandxss here.  The prism files are provided by the prism-loader and thus we don't need the prism dependency explicitly.

**Comment by sirajc** [comment]
Prism loader handles stuff like this very well, even on windows
```
export const basicTsContent = require('!!prismjs?lang=typescript!./basic.alert.ts');
export const basicHtmlContent = require('!!prismjs?lang=markup!./basic.alert.html');
```

Its just this line `import 'prismjs/themes/prism-okaidia.css';` We are directly importing the css in `app.component.ts` this is theme file for prism. Webpack will go to `node_modules/prismjs/themes` to look for this css, but cannot find it.

@icfantv To be sure its Windows only issue, can you check if you have `prismjs` folder in `node_modules`

**Comment by icfantv** [comment]
@sirajc, I'm getting a little frustrated here as I feel like I'm doing your work for you.

prismjs is a dependency of the prism-loader and as such it gets installed when prism-loader gets installed.  see below for the screenshot.

<img width="728" alt="screen shot 2016-06-28 at 20 38 30" src="https://cloud.githubusercontent.com/assets/370054/16438879/60063f60-3d70-11e6-9a1a-0835392cb5e6.png">

**Comment by sirajc** [comment]
I simply don't understand, that y are you not getting the point I am mentioning
Even for me the dependency got installed, See this
```
prismjs-loader@0.0.3 node_modules\prismjs-loader
├── loader-utils@0.2.14 (object-assign@4.1.0, big.js@3.1.3, json5@0.5.0, emojis-list@1.0.3)
└── prismjs@1.4.1
```

But you should also know that the package dependency goes into its own node_modules folder and not the root node_modules folder, See this
![image](https://cloud.githubusercontent.com/assets/5340498/16438986/d98711b4-3dd1-11e6-9539-3f229bb13944.png)

There is no way webpack is going deep into each packages node_modules, Hope you understand that and its clear now

There are chances that you have `prismjs` in your root node_module folder just because you might have ran `npm install prismjs` sometime without having used `--save-dev` and that dependency never got added in package.json

**Comment by icfantv** [comment]
@sirajc, i'm not really sure how else I can explain this.  the only `node_modules` directory i have is the one in `core` and i've blown it away and redone `npm install` several times over the past few days as I flesh out the demo site issues.

i've asked others on the team if they have access to a windows machine and can try and reproduce the issue.  in the interim, it sounds like you have a workaround to keep you moving.  until we can decide on how to proceed, i would encourage you to continue with that.  thanks.

**Comment by sirajc** [comment]
Well no need to try to reproduce this, I am happy with the workaround I have. and if anyone is facing the same issue they ll get to know what needs to be done. Feel free to close #290 if you think adding one dependency is of no value, if it solves issues for few

Talking about the note on your frustration, Here is what I have to say

![image](https://cloud.githubusercontent.com/assets/5340498/16439390/7bdf9474-3dd5-11e6-93b4-4c316ec698ea.png)

I myself maintain projects in Angular2 (https://github.com/sirajc/angular2-bs4-navbar & https://github.com/sirajc/angular2-starter), so I know the basics and wouldn't have troubled you if it was really a user or windows issue. That dependency needs to be there. Rest is your choice, after all you are the "core team member"

**Comment by icfantv** [comment]
@sirajc my frustration is not with you contributing nor did I ever suggest that your contributions weren't welcome - and apologies if i made you feel that way.  we've talked about accepting well-written contributions from the community and indeed have removed its antithesis from our README.

my frustration is that I felt you were too quick to blame the project config for your issues and attempted to solve the issue by just treating the symptoms and not getting at the root cause - and indeed submitted PRs to treat those symptoms.

i hope that makes sense.

**Comment by sirajc** [comment]
Agreed, it makes sense, this is not influencing anything about contribution.

I have raised few issues through this issues
1. Related to Typescript version - and you **fixed** that by updating package.json
2. Exclusion of node_modules - and you **fixed** that by adding it in tsconfig.json
3. Exclusion of node_modules actually fixed angular and rxjs issues thrown by webpack
4. css for prismjs is the only item left and we have PR open for that, which may or may not get merged

So All in all, I did tried to contribute and didn't blame project or anyone. I hope that makes sense as well
